server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -260,6 +260,13 @@
         settings -> Boolean.toString(IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(settings).onOrAfter(LegacyESVersion.V_7_0_0)),
         Property.IndexScope, Property.Final);
 
+    /**
+     * Specifies if the index translog should prune based on retention leases.
+     */
+    public static final Setting<Boolean> INDEX_TRANSLOG_RETENTION_LEASE_PRUNING_ENABLED_SETTING =
```

Comment:
Add a validation on this setting when soft deletes is enabled.

Reply:
Done